### PR TITLE
Add support for HOSTNAME diagnostics message

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:** 

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Add host name error messages/warnings to the Pi-hole diagnostics system

Screenshot from 2020-05-23 11-20-53

**How does this PR accomplish the above?:**

Add front end support for https://github.com/pi-hole/FTL/pull/780

**What documentation changes (if any) are needed to support this PR?:**

None

### CODE REVIEW

File: scripts/pi-hole/js/messages.js [hunk 1]
```diff
@@ -57,6 +57,24 @@ function renderMessage(data, type, row) {
         "</pre> to get the group configuration for this client."
       );
 
+    case "HOSTNAME":
+      var hint = "";
+      for (var i = 0; i < row.blob2 + row.message.length + 2; i++) hint += " ";
```
Comment (XhmikosR) [inline scripts/pi-hole/js/messages.js]: Not sure this is readable, better break it to a new line

File: scripts/pi-hole/js/messages.js [hunk 2]
```diff
@@ -57,6 +57,24 @@ function renderMessage(data, type, row) {
...
+      return (
+        "Host names contains invalid character <code> " +
+        decodeURIComponent(escape(row.blob1))[row.blob2] +
+        "</code>:<pre>" +
+        hint +
+        "↓\n" +
```
Comment (XhmikosR) [inline scripts/pi-hole/js/messages.js]: Might be worth looking into using the HTML entities for the symbols here. We don't use Unicode symbols in JS AFAICT.

File: scripts/pi-hole/js/messages.js [hunk 3]
```diff
@@ -57,6 +57,24 @@ function renderMessage(data, type, row) {
...
+        row.message +
+        ": " +
+        decodeURIComponent(escape(row.blob1)) +
+        "\n" +
+        hint +
+        "↑" +
```
Comment (XhmikosR) [inline scripts/pi-hole/js/messages.js]: This can be combined with the next line

File: scripts/pi-hole/js/messages.js (hunk 1 above)
Comment (DL6ER) [reply scripts/pi-hole/js/messages.js]: `xo` asked me to do it this way. I had it in two lines, initially

File: scripts/pi-hole/js/messages.js (hunk 3 above)
Comment (DL6ER) [reply scripts/pi-hole/js/messages.js]: Again, this is how `xo` wants it

File: scripts/pi-hole/js/messages.js (hunk 1 above)
Comment (XhmikosR) [reply scripts/pi-hole/js/messages.js]: You need brackets.

File: scripts/pi-hole/js/messages.js (hunk 3 above)
Comment (XhmikosR) [reply scripts/pi-hole/js/messages.js]: You can just join the strings is what I mean.

File: scripts/pi-hole/js/messages.js (hunk 2 above)
Comment (DL6ER) [reply scripts/pi-hole/js/messages.js]: Can HTML entities be used inside `<pre>`?

Comment (XhmikosR) [reply scripts/pi-hole/js/messages.js]: If it's not escaped it's not valid IIRC. But anyway just give it a go.

File: scripts/pi-hole/js/messages.js (hunk 1 above)
Comment (DL6ER) [reply scripts/pi-hole/js/messages.js]: I will replace this by a shorter method.

File: scripts/pi-hole/js/messages.js [hunk 4]
```diff
@@ -57,6 +57,22 @@ function renderMessage(data, type, row) {
         "</pre> to get the group configuration for this client."
       );
 
+    case "HOSTNAME":
+      var hint = ' '.repeat(row.blob2 + row.message.length + 2);
```
Comment (XhmikosR) [inline scripts/pi-hole/js/messages.js]: This is also incompatible with IE

Comment (DL6ER) [reply scripts/pi-hole/js/messages.js]: So let's use the one-line `for` loop. This isn't a complicated thing, it doesn't warrant indentation and brackets.

**Comment by DL6ER** [comment]
IE compatibility should be restored + the branch is rebased onto latest `devel`.

**Comment by XhmikosR** [comment]
I don't see why this can't work?

```js
for (var i = 0; i < row.blob2 + row.message.length + 2; i++) {
  hint += " ";
}
```

**Comment by PromoFaux** [comment]
Just chucking this in the ring:

```js
var hint = new Array(row.blob2 + row.message.length + 3).join(" ");
```

**Comment by DL6ER** [comment]
@XhmikosR This works as well, however, it is really verbose for a very simple task. It adheres to our style criteria and is ready to merge IMO. @PromoFaux We can use this as well, can you commit/ppush it?

It is also worth pointing out that the failed node tests do not have to do with the changes in this PR.

**Comment by XhmikosR** [comment]
I honestly don't see how it's more verbose... I mean, the whole codebase is verbose :P

Anyway, prettier will go away soon.

File: scripts/pi-hole/js/messages.js (hunk 1 above)
Comment (PromoFaux) [reply scripts/pi-hole/js/messages.js:62]:
```suggestion
      var hint = new Array(row.blob2 + row.message.length + 3).join(" ");
```

File: scripts/pi-hole/js/messages.js [hunk 5]
```diff
@@ -57,6 +57,23 @@ function renderMessage(data, type, row) {
         "</pre> to get the group configuration for this client."
       );
 
+    case "HOSTNAME":
+      var hint = "";
```
Comment (PromoFaux) [inline scripts/pi-hole/js/messages.js]:
```suggestion

```

**Review by PromoFaux — CHANGES_REQUESTED** [review]

File: scripts/pi-hole/js/messages.js [hunk 6]
```diff
@@ -57,6 +57,23 @@ function renderMessage(data, type, row) {
...
+      for (var i = 0; i < row.blob2 + row.message.length + 2; i++) hint += " ";
```
Comment (PromoFaux) [inline scripts/pi-hole/js/messages.js]:
```suggestion
      var hint = new Array(row.blob2 + row.message.length + 3).join(" ");
```

File: scripts/pi-hole/js/messages.js [hunk 7]
```diff
@@ -57,6 +57,22 @@ function renderMessage(data, type, row) {
         "</pre> to get the group configuration for this client."
       );
 
+    case "HOSTNAME":
+      var hint = new Array(row.blob2 + row.message.length + 3).join(" ");
+      return (
+        "Host names contains invalid character <code> " +
```
Comment (XhmikosR) [inline scripts/pi-hole/js/messages.js]: Typo `contains` :)

Comment (XhmikosR) [reply scripts/pi-hole/js/messages.js]: Also should probably be `hostnames`?

Comment (PromoFaux) [reply scripts/pi-hole/js/messages.js]: If that's all you can come up with, then today I have successfully rebased something ;) I was just about to ping you and get you to doublecheck before  I did anything else.

Though I need to make sure I've not lost some other stuff first...

Comment (PromoFaux) [reply scripts/pi-hole/js/messages.js]: Wait. There is no typo in `contains`...

Comment (PromoFaux) [reply scripts/pi-hole/js/messages.js]: Agree with the latter though

Comment (XhmikosR) [reply scripts/pi-hole/js/messages.js]: It's plural, so it should be contain?

Comment (XhmikosR) [reply scripts/pi-hole/js/messages.js]: Well, I don't quite get the Array solution there TBH, but the rebase seems to be good AFAICT :)

File: scripts/pi-hole/js/messages.js [hunk 8]
```diff
@@ -57,6 +57,22 @@ function renderMessage(data, type, row) {
         "</pre> to get the group configuration for this client."
       );
 
+    case "HOSTNAME":
+      var hint = new Array(row.blob2 + row.message.length + 3).join(" ");
+      return (
+        "Hostname contains invalid character <code> " +
```
Comment (XhmikosR) [inline scripts/pi-hole/js/messages.js]: Now that I see this, maybe remove the space since it's inside `<code>`?

Comment (PromoFaux) [reply scripts/pi-hole/js/messages.js]: One sec, I'm going to need to reproduce the error again to get it to display

**Review by PromoFaux — APPROVED** [review]

**Pull request merged**